cmake/thirdparty/get_gtest.cmake

```diff
@@ -0,0 +1,43 @@
+# =============================================================================
```

Comment:
Should the `get_gtest.cmake` and `get_gbench.cmake` be promoted to the `rapids-cmake` repo? I'd think they would be duplicated across multiple RAPIDS repos if not.

Reply:
I think that gtest, gbench, and spdlog are all very resonable candiates to getting
moved over to rapids-cmake. In general I am hestitant to make rapids-cmake a package
manager but for core common dependencies it does make sense. I think thrust could also be included but we need to discuss that a bit more, since cudf uses a patched version of thrust ( do we want all of rapids to also do that? )

Most of the questions will be around what does the API look like for this. I think we would want to offer an API that offers very little customization ( around which version, flags, etc ) since `rapids_cpm_find` offers that. Instead we would be looking at a fairly closed/narrow API like the following: 


```
  rapids_cpm_<PackageName>( [BUILD_EXPORT_SET <export-name>]
                            [INSTALL_EXPORT_SET <export-name>]                  
                          )
```

Which would simplify `rmm` usage to:
```
rapids_cpm_gtest()
rapids_cpm_benchmark()
rapids_com_spdlog(BUILD_EXPORT_SET rmm-exports INSTALL_EXPORT_SET rmm-exports)
```

Reply:
That looks good. However I guess this shouldn't block the present PR. Agree?

Reply:
I agree.

I have added a feature request to rapids-cmake for this: https://github.com/rapidsai/rapids-cmake/issues/32